### PR TITLE
Clojure lint

### LINKED ISSUE #111: Clojure support
Using https://github.com/borkdude/clj-kondo

### PR DESCRIPTION
**Pull request by admiralAwkbar**
Fixes #111 by adding the clj-kondo CLI to the runtime environment and binding the following extensions to the linter: clj, cljc, cljs, edn.

This closes #197

### CODE REVIEW

**Comment by admiralAwkbar** [comment]
@goedelsoup Can you take a look at the failure?

Error seems to be from: `ERROR: Linter CMD:[clj-kondo --config /action/lib/.automation/.clj-kondo/config.edn ./clojure/clojure_bad_1.clj]
`

**Comment by admiralAwkbar** [comment]
@goedelsoup so I found the issue with the command but this brought up an interesting behavior.

When a file is listed, its exit code is not `0` for success...
```bash
bash-5.0# clj-kondo --run-as-prod --config /action/lib/.automation/.clj-kondo/config.edn --lint good.clj
good.clj:11:11: warning: unused binding z
linting took 13ms, errors: 0, warnings: 1
bash-5.0# echo $?
2
```

I have read that updating the config could help, but I don't know Clojure that well... Can you update the config to help it exit with `0` for success, and `>0` for errors?

**Comment by danielcompton** [comment]
> I have read that updating the config could help, but I don't know Clojure that well... Can you update the config to help it exit with 0 for success, and >0 for errors?

I've fixed this, it was the GOOD file that had a lint warning left in it. I've removed that now.

File: TEMPLATES/.clj-kondo/config.edn [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{:linters {:unresolved-symbol {:exclude [(compojure.api.sweet/defroutes)]}
```
[the remark this answers is not in the document]
Comment (borkdude) [reply TEMPLATES/.clj-kondo/config.edn:1]: clj-kondo's config depends very much on the project, there is no general config that will satisfy all projects. So an empty `{}` makes sense.

**Comment by borkdude** [comment]
@danielcompton clj-kondo is able to output in a format that can be used for annotations.
See here: https://github.com/borkdude/clj-kondo/blob/master/doc/ci-integration.md#linter-output-integration

**Pull request merged**